app/models/manageiq/providers/nuage/network_manager/cloud_tenant.rb

```diff
@@ -1,4 +1,10 @@
 class ManageIQ::Providers::Nuage::NetworkManager::CloudTenant < ::CloudTenant
+  has_many   :security_groups, :dependent => :destroy
+  has_many   :cloud_subnets,   :dependent => :destroy
+  has_many   :network_ports,   :dependent => :destroy
```

Comment:
so, this should remove all network_ports, is the case that these are some owned by other tenant?

Reply:
ok, makes sense. I don't think we have a better way to do this now

Reply:
btw. nuage doesn't send events for all the objects that are deleted?

Reply:
As far as I've tested it doesn't. It only sends event for explicitly deleted entity while it silently removes its children..

Reply:
hm strange, in other providers, we usually get event per each entity